apis/projectcontour/v1/httpproxy.go

```diff
@@ -324,6 +328,12 @@ type RetryPolicy struct {
 	// PerTryTimeout specifies the timeout per retry attempt.
 	// Ignored if NumRetries is not supplied.
 	PerTryTimeout string `json:"perTryTimeout,omitempty"`
+	// RetryOn specifies the conditions on which to retry a request.
+	// +optional
```

Comment:
This is generated into API reference documentation, so we should enumerate and document the values (the generator doesn't look at the kubebuilder annotations). This comment an be markdown, so perhaps the balance is to add a list to enumerate the options, then link to Envoy docs for detailed semantics.